tools/server/server.cpp

```diff
@@ -3380,7 +3491,7 @@ struct server_context {
                 slot.n_draft_accepted += ids.size() - 1;
 
                 slot.cache_tokens.push_back(id);
-                slot.cache_tokens.insert(slot.cache_tokens.end(), ids.begin(), ids.end() - 1);
+                slot.cache_tokens.insert(ids);
```

Comment:
This fixes the speculative decoding performance:

```suggestion
                slot.cache_tokens.insert({ids.begin(), ids.end() - 1});
```